python/recipe.hpp

```diff
@@ -123,48 +123,66 @@ class py_recipe_shim: public arb::recipe {
     py_recipe_shim(std::shared_ptr<py_recipe> r): impl_(std::move(r)) {}
 
     arb::cell_size_type num_cells() const override {
-        return impl_->num_cells();
+        return try_catch_pyexception(
+                    [&](){ return impl_->num_cells(); },
+                    "A Python error in num_cells() on a different thread");
```

Comment:
This error message is put into a `runtime_error` that is used to indicate that a Python error was generated by an earlier Python callback. It will (hopefully) never be seen by the user, because the wrapping code will ensure that the Python error is thrown back into the calling Python context.
So, this could be replaced with a common short error string like `"Python error already thrown"`

Reply:
Done (in .hpp stored as `const char*` for readability)